### PR TITLE
feat(v2): blog support date front matter

### PR DESCRIPTION
**Pull request by wenerme**
## Motivation

Use a better slugify function that doesn't rely on a specific file extension.

### Have you read the [Contributing Guidelines on pull requests](https://github.com/facebook/Docusaurus/blob/master/CONTRIBUTING.md#pull-requests)?

yes

## Test Plan

(Write your test plan here. If you changed any code, please provide us with clear instructions on how you verified your changes work. Bonus points for screenshots and videos!)

add date in blog doc

## Related PRs

(If this PR adds or changes functionality, please take some time to update the docs at https://github.com/facebook/docusaurus, and link to your PR here.)

n/a

### CODE REVIEW

**Comment by wenerme** [comment]
@endiliey how to test? just change current website-1.x/blog ? I don't find any tests about the blog.

**Comment by endiliey** [comment]
write at least unit tests for your newly written code especially on function. 

see how we test docs
https://github.com/facebook/docusaurus/tree/master/packages/docusaurus-plugin-content-docs-legacy/src/__tests__

**Comment by wenerme** [comment]
I do found tests about docs. I'll try to make a test about blogs

**Comment by wenerme** [comment]
test has been added.

**Comment by wenerme** [comment]
fixed and add a test

File: packages/docusaurus-plugin-content-blog/src/index.js [hunk 1]
```diff
@@ -69,30 +71,40 @@ module.exports = function(context, opts) {
...
+            const [, , dateString, name] = match;
+            date = new Date(dateString);
+            linkName = name;
+          }
+          // prefer usedefined date
+          if (frontMatter.date) {
+            date = new Date(frontMatter.date);
+          }
+          // use file create time for blog
+          date = date || (await fs.fstat(source)).ctime;
```
Comment (endiliey) [inline packages/docusaurus-plugin-content-blog/src/index.js]: ctime will always change depending on when the file was changed

maybe birthtime (file creation) is better ?

Comment (endiliey) [reply packages/docusaurus-plugin-content-blog/src/index.js]: https://nodejs.org/api/fs.html#fs_stats_ctime

Comment (wenerme) [reply packages/docusaurus-plugin-content-blog/src/index.js]: yes, misunderstanding ctime, used to thought is abbr for creation time.

**Comment by wenerme** [comment]
change to birthtime, add test for no time, fix the link issues

**Comment by wenerme** [comment]
dose this also need to update changelog-2x.md ?

**Comment by endiliey** [comment]
yup

**Comment by endiliey** [comment]
friendly ping to resolve conflict

**Comment by wenerme** [comment]
Why I don't recv any notification, 😑

**Pull request merged**